x11-terms/ghostty/ghostty-1.0.0.ebuild

```diff
@@ -0,0 +1,164 @@
+	[wayland_headers-1220b3164434d2ec9db146a40bf3a30f490590d68fa8529776a3138074f0da2c11ca.tar.gz]='https://github.com/mitchellh/wayland-headers/archive/5f991515a29f994d87b908115a2ab0b899474bd1.tar.gz'
+	[wuffs-12200984439edc817fbcbbaff564020e5104a0d04a2d0f53080700827052de700462.tar.gz]='https://github.com/google/wuffs/archive/refs/tags/v0.4.0-alpha.8.tar.gz'
+	[x11_headers-122089c326186c84aa2fd034b16abc38f3ebf4862d9ae106dc1847ac44f557b36465.tar.gz]='https://github.com/mitchellh/x11-headers/archive/2ffbd62d82ff73ec929dd8de802bc95effa0ef88.tar.gz'
+	[xcode_frameworks-12202adbfecdad671d585c9a5bfcbd5cdf821726779430047742ce1bf94ad67d19cb.tar.gz]='https://github.com/mitchellh/xcode-frameworks/archive/69801c154c39d7ae6129ea1ba8fe1afe00585fc8.tar.gz'
+	[z2d-12201f0d542e7541cf492a001d4d0d0155c92f58212fbcb0d224e95edeba06b5416a.tar.gz]='https://github.com/vancluever/z2d/archive/4638bb02a9dc41cc2fb811f092811f6a951c752a.tar.gz'
+	[zf-1220edc3b8d8bedbb50555947987e5e8e2f93871ca3c8e8d4cc8f1377c15b5dd35e8.tar.gz]='https://github.com/natecraddock/zf/archive/ed99ca18b02dda052e20ba467e90b623c04690dd.tar.gz'
+	[zg-122055beff332830a391e9895c044d33b15ea21063779557024b46169fb1984c6e40.tar.gz]='https://codeberg.org/atman/zg/archive/v0.13.2.tar.gz'
+	[zig-objc-1220e17e64ef0ef561b3e4b9f3a96a2494285f2ec31c097721bf8c8677ec4415c634.tar.gz]='https://github.com/mitchellh/zig-objc/archive/9b8ba849b0f58fe207ecd6ab7c147af55b17556e.tar.gz'
+	[zig_js-12205a66d423259567764fa0fc60c82be35365c21aeb76c5a7dc99698401f4f6fefc.tar.gz]='https://github.com/mitchellh/zig-js/archive/d0b8b0a57c52fbc89f9d9fecba75ca29da7dd7d1.tar.gz'
+	[zigimg-1220dd654ef941fc76fd96f9ec6adadf83f69b9887a0d3f4ee5ac0a1a3e11be35cf5.tar.gz]='https://github.com/zigimg/zigimg/archive/3a667bdb3d7f0955a5a51c8468eac83210c1439e.tar.gz'
+	[ziglyph-12207831bce7d4abce57b5a98e8f3635811cfefd160bca022eb91fe905d36a02cf25.tar.gz]='https://deps.files.ghostty.org/ziglyph-b89d43d1e3fb01b6074bc1f7fc980324b04d26a5.tar.gz'
+	[zlib-1220fed0c74e1019b3ee29edae2051788b080cd96e90d56836eea857b0b966742efb.tar.gz]='https://github.com/madler/zlib/archive/refs/tags/v1.3.1.tar.gz'
+)
+
+ZIG_SLOT="0.13"
+inherit zig xdg-utils
+
+SRC_URI="
+	https://release.files.ghostty.org/${PV}/ghostty-source.tar.gz -> ${P}.tar.gz
+	${ZBS_DEPENDENCIES_SRC_URI}
+"
+S="${WORKDIR}/ghostty-source"
+
+LICENSE="
+	Apache-2.0 BSD BSD-2 BSD-4 Boost-1.0 MIT MPL-2.0
+	!system-freetype? ( || ( FTL GPL-2+ ) )
+	!system-harfbuzz? ( Old-MIT ISC icu )
+	!system-libpng? ( libpng2 )
+	!system-zlib? ( ZLIB )
+"
+SLOT="0"
+KEYWORDS="~amd64"
+
+# TODO: simdutf integration (missing Gentoo version)
+# TODO: spirv-cross integration (missing Gentoo package)
+# TODO: glfw integration (no option from upstream)
+BDEPEND="
+	adwaita? ( gui-libs/libadwaita:1= )
+	gtk? ( gui-libs/gtk:4= )
+	man? ( virtual/pandoc )
+
+	system-fontconfig? ( >=media-libs/fontconfig-2.14.2:= )
```

Comment:
We generally do [unconditional system libraries](https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies) unless there's a good reason not to, e.g. upstream sometimes modify the bundled copy.

Reply:
Hmm, this is the template that `zig-ebuilder` started me off with. I could certainly make system packages unconditional, although I'm not sure if I can then remove the Zig dependencies without upsetting the build system.

Reply:
1) Sometimes they do, when packaging to use `build.zig` they also remove some files, patch to be compilable with Clang and so on (but it seems like patch part does not apply to any Ghostty dependency as far as I saw)
2) It is useful for reproducibility and when someone needs them to troubleshoot with and without bundled dependencies
3) Users can't override to disable or enable part of them globally, because `zig build` will fail with:
```
system library name not recognized by build script: 'some_library'
  access the help menu with 'zig build -h'
```
so if they want this configured (to skip dependency etc.) they would need `package.env`, I think USE-flag would be more appropriate here

Reply:
Alright, keeping the USE flags for now then.